app/components/button/index.tsx

```diff
@@ -2,12 +2,15 @@
 // See LICENSE.txt for license information.
 
 import React, {useMemo} from 'react';
-import {StyleProp, Text, TextStyle, ViewStyle} from 'react-native';
+import {StyleProp, StyleSheet, Text, TextStyle, View, ViewStyle} from 'react-native';
 import RNButton from 'react-native-button';
 
+import CompassIcon from '@components/compass_icon';
 import {buttonBackgroundStyle, buttonTextStyle} from '@utils/buttonStyles';
 
-type Props = {
+type ConditionalProps = | {iconName: string; iconSize: number} | {iconName?: never; iconSize?: never}
```

Comment:
Does the first `|` serve any purpose?

Reply:
Yes, this is what it makes it to be conditional